components/NavBar.tsx

```diff
@@ -0,0 +1,69 @@
+import HomeIcon from "./HomeIcon";
+import ToggleStormModeStatusButton from "./ToggleStormModeStatusButton";
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import AdminDashboardIcon from "./AdminDashboardIcon";
+import useAdmin from "../hooks/useAdmin";
+import ToggleStormModeConfirmationModal from "./ToggleStormModeConfirmationModal";
+import { useEffect, useState } from "react";
+import useToggleStormModeStatus from "../hooks/useToggleStormModeStatus";
+
+type Props = {
+  stormModeStatus: boolean;
+  adminPassphrase: string;
+};
+
+const Container = styled.div`
+  position: fixed;
+  top: 0;
+  display: flex;
+  justify-content: flex-end;
+  width: 100%;
+`;
+
+function NavBar({ stormModeStatus, adminPassphrase }: Props) {
+  const [confirmationModalOpen, setConfirmationModalOpen] = useState(false);
+  const [status, setStatus] = useState(stormModeStatus);
+  const { data } = useToggleStormModeStatus(adminPassphrase);
+
+  useEffect(() => {
+    setStatus(data);
+  }, [data]);
```

Comment:
Since you're just copying `data` into `status`, I think this is extraneous. You can remove this `useEffect` block and `status`/`setStatus`, and use `data` + `mutate` from `useToggleStormModeStatus` instead.

Reply:
updated @zxqx